test/src/test/java/org/corfudb/runtime/collections/CorfuStoreTest.java

```diff
@@ -0,0 +1,310 @@
+
+        corfuStore.tx(nsxManager).delete(tableName, key1).commit();
+        assertThat(corfuStore.getTable(nsxManager, tableName).get(key1)).isNull();
+        expectedVersion = 1L;
+
+        corfuStore.tx(nsxManager)
+                .update(tableName,
+                        key1,
+                        EventInfo.newBuilder().setName("def").build(),
+                        null)
+                .commit();
+        assertThat(corfuStore.getTable(nsxManager, tableName).get(key1).getMetadata())
+                .isEqualTo(ManagedResources.newBuilder().setVersion(expectedVersion).build());
+    }
+
+    @Test
+    public void checkNullMetadataTransactions() throws Exception {
+
+        // Get a Corfu Runtime instance.
+        CorfuRuntime corfuRuntime = getDefaultRuntime();
+
+        // Creating Corfu Store using a connected corfu client.
+        CorfuStore corfuStore = new CorfuStore(corfuRuntime);
+
+        // Define a namespace for the table.
+        final String nsxManager = "nsx-manager";
+        // Define table name.
+        final String tableName = "EventInfo";
+
+        // Create & Register the table.
+        // This is required to initialize the table for the current corfu client.
+        Table<Uuid, EventInfo, ManagedResources> table = corfuStore.createTable(
+                nsxManager,
+                tableName,
+                Uuid.class,
+                EventInfo.class,
+                null,
+                // TableOptions includes option to choose - Memory/Disk based corfu table.
+                TableOptions.builder().build());
+
+        UUID uuid1 = UUID.nameUUIDFromBytes("1".getBytes());
+        Uuid key1 = Uuid.newBuilder().setMsb(uuid1.getMostSignificantBits()).setLsb(uuid1.getLeastSignificantBits()).build();
+        corfuStore.tx(nsxManager)
+                .create(tableName,
+                        key1,
+                        EventInfo.newBuilder().setName("abc").build(),
+                        null)
+                .commit();
+        assertThat(corfuStore.getTable(nsxManager, tableName).get(key1).getMetadata())
+                .isNull();
+
+        corfuStore.tx(nsxManager)
+                .update(tableName,
+                        key1,
+                        EventInfo.newBuilder().setName("bcd").build(),
+                        ManagedResources.newBuilder().setCreateUser("testUser").setVersion(1L).build())
+                .commit();
+        assertThat(corfuStore.getTable(nsxManager, tableName).get(key1).getMetadata())
+                .isNull();
+
+        corfuStore.tx(nsxManager)
+                .update(tableName,
+                        key1,
+                        EventInfo.newBuilder().setName("cde").build(),
+                        null)
+                .commit();
+        assertThat(corfuStore.getTable(nsxManager, tableName).get(key1).getMetadata())
+                .isNull();
+    }
+
+//    @Test
```

Comment:
Why is this test commented?

Reply:
Thanks Anny, I think there is value in adding this test which demonstrates how to use DynamicMessage. Let me uncomment and update it.